src/frontend/shape/rectangle.rs

```diff
@@ -0,0 +1,42 @@
+#[derive(Clone)]
+pub struct Rectangle {
+    pub x1: f64,
+    pub x2: f64,
+    pub y1: f64,
+    pub y2: f64,
+}
```

Comment:
I'd prefer to have a `Point` structure like
```rust
#[derive(Debug)]
pub struct Point {
    x: f64,
    y: f64,
}
```
in `src/frontend/shape/mod.rs`, then you could write something like:
```rust
#[derive(Clone)]
pub struct Rectangle {
    pub top_left: Point,
    pub bottom_right: Point,
}
```
if I'm understanding the coordination correctly.